File: backend-dotnet/SahayaSetu/Services/IAuthService.cs
using SahayaSetu.DTO.Auth;
using SahayaSetu.DTO.NGO;

namespace SahayaSetu.Services
{
    public interface IAuthService
    {
        Task<AuthResponseDto> RegisterDonorAsync(DonorRegisterDto dto);
        Task<AuthResponseDto> RegisterAdminAsync(AdminRegisterDto dto);
        Task<AuthResponseDto> RegisterNgoAsync(NgoRegisterDto dto);
        Task<AuthResponseDto> LoginUserAsync(LoginDto dto);
        Task<AuthResponseDto> LoginNgoAsync(LoginDto dto);
    }
}

File: backend-java/sahayaSetu/src/main/java/com/sahayaSetu/services/IAuthService.java
package com.sahayaSetu.services;

import com.sahayaSetu.dtos.*;

public interface IAuthService {
    AuthResponseDto registerDonor(DonorRegistrationDto dto);
    AuthResponseDto registerNgo(NgoRegistrationDto dto);
    AuthResponseDto loginUser(LoginDto dto);
    AuthResponseDto loginNgo(LoginDto dto);
    AuthResponseDto registerAdmin(AdminRegistrationDto dto);
}
